quickwit-config/src/index_config.rs

```diff
@@ -126,20 +130,20 @@ impl MergePolicy {
 impl Default for MergePolicy {
     fn default() -> Self {
         Self {
-            demux_factor: Self::default_demux_factor(),
+            __demux_factor_deprecated: serde::de::IgnoredAny,
             merge_factor: Self::default_merge_factor(),
             max_merge_factor: Self::default_max_merge_factor(),
         }
     }
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
 pub struct IndexingSettings {
-    #[serde(default, skip_serializing_if = "is_false")]
-    pub demux_enabled: bool,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub demux_field: Option<String>,
+    #[serde(default, rename = "demux_enabled", skip_serializing)]
+    pub __demux_enabled_deprecated: IgnoredAny,
```

Comment:
what is the benefit of using `IgnoreAny` instead of `#[serde(default, rename = "demux_enabled", skip)]`?